httpcache.go

```diff
@@ -225,17 +225,18 @@ func (t *Transport) RoundTrip(req *http.Request) (resp *http.Response, err error
 					// Associate original request with cloned request so we can refer to
 					// it in CancelRequest()
 					t.setModReq(req, req2)
+					keyReq := req
```

Comment:
What does the name of variable `keyReq` mean? Is it the "request that is the _key_ of the `modReq` map"?

I think this code is quite subtle and not obvious, so an inline comment explaining the purpose of this line would be very helpful. Something like this:

```Go
keyReq := req // Keep a pointer to the original request so we can delete it from the `modReq` map later.
```